std-bits/table/src/main/java/org/enso/table/data/column/storage/StringStorage.java

```diff
@@ -46,6 +50,22 @@ public TextType getType() {
     return type;
   }
 
+  /**
+   * Counts the number of cells in the columns with whitespace. Memoized into the storage for
+   * performance.
+   *
+   * @return the number of cells with whitespace
+   */
+  public Long countUntrimmed() {
+    if (_countLeadingTrailingWhitespace != -1) {
+      return _countLeadingTrailingWhitespace;
+    }
+
+    _countLeadingTrailingWhitespace =
+        CountUntrimmed.compute(this, CountUntrimmed.DEFAULT_SAMPLE_SIZE);
+    return _countLeadingTrailingWhitespace;
+  }
```

Comment:
We have a race here that can lead to computing `CountUntrimmed.compute` twice. Not a big issue, but perhaps we could avoid doing double the work.

Imagine a (quite possible I guess) scenario:
1. We instantiate the storage, this schedules the `CompletableFuture.runAsync(this::countUntrimmed);`. It starts running `CountUntrimmed.compute` in the background.
2. Very soon, before the `CompletableFuture` invocation above finishes, someone else calls `storage.countUntrimmed()`.

In that case, the second call still sees `_countLeadingTrailingWhitespace == -1` and calls `CountUntrimmed.compute` again. Then one and the other finish and both write the result. Thus we ended up computing the costly `CountUntrimmed.compute` twice, and did not get the result any faster. We could have gotten it slower because the background thread may have put additional strain on the CPU (depending how many cores the machine has and how much usage).

Anyway, we could easily improve this.

We could:
1. Replace `long _countLeadingTrailingWhitespace` with `Future<Long> leadingTrailingWhitespaceCount;`
2. In the constructor, call `leadingTrailingWhitespaceCount = CompletableFuture.supplyAsync(() -> CountUntrimmed.compute(this, CountUntrimmed.DEFAULT_SAMPLE_SIZE));
3. Replace code in `countUntrimmed` with just `leadingTrailingWhitespaceCount.get()`.

This way the computation is run exactly once, and the caller of `countUntrimmed` gets it as soon as it is available. The code is also cleaner.

Reply:
Also while at it, regardless if we use `runAsync` like now or `supplyAsync` like I suggest, we should think about what thread pool we are using. I guess perhaps for now using `CompletableFuture`'s `ForkJoinPool.commonPool()` is okay, but once we start doing more computations like that in the background, it may be good to think about setting up a dedicated thread pool for them, so that we can have more control over them, and maybe even do some kind of prioritization.

Reply:
Agree we will want our own executor for this kind of thing later.